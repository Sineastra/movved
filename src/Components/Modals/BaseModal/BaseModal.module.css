.backdrop {
  visibility: visible;
  opacity: 1;
  background: rgba(0, 0, 0, 0.5);
  z-index: 2;
  width: 100%;
  position: fixed;
  top: 0;
  left: 0;
  height: 100vh;
}
.backdrop.hiddenBackdrop {
  visibility: hidden;
  opacity: 0;
  transition: all 0.3s ease;
}
.backdrop > .wrapper {
  position: absolute;
  top: 50%;
  transform: translate(0, -50%);
  left: 5vw;
  width: 90vw;
  height: 400px;
  z-index: 4;
  background: var(--theme-main-color_2);
  border-radius: 5px;
  transition: all 0.3s linear;
  overflow-y: auto;
  display: flex;
  flex-direction: column;
  justify-content: initial;
  align-items: center;
  align-content: initial;
}
.backdrop > .wrapper.hiddenWrapper {
  opacity: 0;
  visibility: hidden;
  transition: all 0.3s linear;
}
.backdrop > .wrapper > .headingSection {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  align-content: initial;
  padding: 15px;
  vertical-align: middle;
  font-family: "Nunito", sans-serif;
  height: 55px;
  width: 100%;
}
.backdrop > .wrapper > .headingSection > div {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  align-content: center;
}
.backdrop > .wrapper > .headingSection > h3 {
  margin: 0;
}
.backdrop > .wrapper > hr {
  width: 95%;
}
.backdrop > .wrapper > .contentSection {
  padding-top: 5px;
  height: 325px;
  width: 95%;
  text-align: center;
}

/*# sourceMappingURL=BaseModal.module.css.map */
