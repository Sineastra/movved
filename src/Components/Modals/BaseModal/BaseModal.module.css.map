{"version":3,"sourceRoot":"","sources":["BaseModal.module.scss","../../../_scssPartials/mixins/_mixins.scss"],"names":[],"mappings":"AAEA;EACI;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;;AAEA;EACI;EACA;EACA;;AAGJ;EACI;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;;AAEA;EACI;EACA;EACA;;AAGJ;ECrCJ;EACA,gBDqC6B;ECpC7B,iBDoCkC;ECnClC,aDmCiD;EClCjD,eALwF;EDwChF;EACA;EACA;;AAEA;EC3CR;EACA,gBAM4B;EAL5B,iBAMiC;EALjC,aAKyC;EAJzC,eAIiD;;ADuCzC;EACI;;AAIR;EACI;;AAGJ;ECxDJ;EACA,gBAM4B;EAL5B,iBAMiC;EALjC,aAKyC;EAJzC,eAIiD;EDkDzC","file":"BaseModal.module.css"}