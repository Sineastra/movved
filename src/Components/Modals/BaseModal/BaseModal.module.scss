@import "../../../_scssPartials/mixins/mixins";

.backdrop {
    visibility: visible;
    opacity: 1;
    background: rgba(0, 0, 0, 0.5);
    z-index: 2;
    width: 100%;
    position: fixed;
    top: 0;
    left: 0;
    height: 100vh;

    &.hiddenBackdrop {
        visibility: hidden;
        opacity: 0;
        transition: all 0.3s ease;
    }

    & > .wrapper {
        position: absolute;
        top: 50%;
        transform: translate(0, -50%);
        left: 5vw;
        width: 90vw;
        height: 400px;
        z-index: 4;
        background: var(--theme-main-color_2);
        border-radius: 5px;
        transition: all 0.3s linear;
        overflow-y: auto;
        @include displayFlex(column, initial, center);

        &.hiddenWrapper {
            opacity: 0;
            visibility: hidden;
            transition: all 0.3s linear;
        }

        & > .headingSection {
            @include displayFlex(row, space-between, center);
            padding: 15px;
            vertical-align: middle;
            font-family: 'Nunito', sans-serif;
            height: 55px;
            width: 100%;

            & > div {
                @include centeredFlex()
            }

            & > h3 {
                margin: 0;
            }
        }

        & > hr {
            width: 95%;
        }

        & > .contentSection {
            padding-top: 5px;
            height: 325px;
            width: 95%;
            text-align: center;
        }
    }
}