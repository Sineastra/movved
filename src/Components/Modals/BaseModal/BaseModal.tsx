import styles from "./BaseModal.module.scss"
import React, { BaseSyntheticEvent, ReactElement } from "react"
import { RiCloseLine } from "react-icons/ri"
import { IconContext } from "react-icons"


interface propsIntF {
	heading: string,
	showModal: boolean,
	icon: ReactElement,
	setShowModal: (v: boolean) => void,
	modalClassName?: string,
}
const BaseModal: React.FC<propsIntF> = ({ heading, showModal, setShowModal, children, icon, modalClassName = '' }) => {

	const hideModalBackdrop = (e: BaseSyntheticEvent) => {
		const isNotCloseBtn = e.target.tagName !== "svg" && e.target.tagName !== "path"

		if (isNotCloseBtn && e.target.className.trim() === styles.backdrop) {
			setShowModal(false)
		}
	}

	return (
		<div className={ `${ styles.backdrop } ${ showModal ? '' : styles.hiddenBackdrop }` }
		     role="dialog"
		     onClick={ hideModalBackdrop }>
			<div className={ `${ styles.wrapper } ${ showModal ? '' : styles.hiddenWrapper } ${ modalClassName }` }
			     title="dialog window">
				<section className={ styles.headingSection }>
					<IconContext.Provider value={ { size: "1.4rem" } }>
						{ icon }
						<h3>{ heading }</h3>
						<div onClick={ () => setShowModal(false) }>
							<RiCloseLine role="graphics-document" aria-label="close-modal"
							             style={ { cursor: "pointer" } }/>
						</div>
					</IconContext.Provider>
				</section>
				<hr/>
				<section className={ styles.contentSection }>
					{ children }
				</section>
			</div>
		</div>
	)
}

export default BaseModal