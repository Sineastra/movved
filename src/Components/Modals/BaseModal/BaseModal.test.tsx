import renderer from "react-test-renderer"
import { BrowserRouter } from "react-router-dom"
import BaseModal from "./BaseModal"
import { render, screen } from "@testing-library/react"
import user from "@testing-library/user-event"


const setShowModal = jest.fn()

describe("--> Testing BaseModal", () => {
	const renderScreen = () =>
		render(
			<BrowserRouter>
				<BaseModal heading="a" showModal={ true } setShowModal={ setShowModal } icon={ <div></div> }>
					<div>a</div>
				</BaseModal>
			</BrowserRouter>,
		)
	it("renders correctly (snapshot test)", () => {
		const tree = renderer.create(
			<BrowserRouter>
				<BaseModal heading="a" showModal={ true } setShowModal={ () => {} } icon={ <div></div> }>
					<div>a</div>
				</BaseModal>
			</BrowserRouter>,
		).toJSON()

		expect(tree).toMatchSnapshot()
	})
	it("correctly calls close modal callback on press on backdrop", async () => {
		renderScreen()
		const backdrop = screen.getByRole('dialog')

		await user.click(backdrop)

		expect(setShowModal).toHaveBeenCalledTimes(1)
	})
	it("does not call close modal callback on pressing the dialog window", async () => {
		renderScreen()
		const dialogWindow = screen.getByTitle(/dialog window/i)

		await user.click(dialogWindow)

		expect(setShowModal).toHaveBeenCalledTimes(0)
	})
	it("does call close modal callback when pressing the close icon", async () => {
		renderScreen()
		const closeIcon = screen.getByRole("graphics-document", { name: 'close-modal' })

		await user.click(closeIcon)

		expect(setShowModal).toHaveBeenCalledTimes(1)
	})
})