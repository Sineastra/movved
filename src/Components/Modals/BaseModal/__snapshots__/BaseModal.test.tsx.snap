// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`--> Testing BaseModal renders correctly (snapshot test) 1`] = `
<div
  className="backdrop "
  onClick={[Function]}
  role="dialog"
>
  <div
    className="wrapper  "
    title="dialog window"
  >
    <section
      className="headingSection"
    >
      <div />
      <h3>
        a
      </h3>
      <div
        onClick={[Function]}
      >
        <svg
          aria-label="close-modal"
          fill="currentColor"
          height="1.4rem"
          role="graphics-document"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1.4rem"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M12 10.586l4.95-4.95 1.414 1.414-4.95 4.95 4.95 4.95-1.414 1.414-4.95-4.95-4.95 4.95-1.414-1.414 4.95-4.95-4.95-4.95L7.05 5.636z"
            />
          </g>
        </svg>
      </div>
    </section>
    <hr />
    <section
      className="contentSection"
    >
      <div>
        a
      </div>
    </section>
  </div>
</div>
`;
